packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,254 @@
+  }
+
+  /**
+   * See {IGachaLazyClaim-mintReserve}.
+   */
+  function mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) external payable override {
+    if (Address.isContract(msg.sender)) revert IGachaLazyClaim.CannotMintFromContract();
+    Claim storage claim = _getClaim(creatorContractAddress, instanceId);
+    // Checks for reserving
+    if (claim.startDate > block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (claim.endDate > 0 && claim.endDate <= block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (msg.value != (claim.cost + MINT_FEE) * mintCount) revert IGachaLazyClaim.InvalidPayment();
+    uint32 amountAvailable = claim.totalMax - claim.total;
+    if (amountAvailable == 0) revert IGachaLazyClaim.ClaimSoldOut();
+
+    // calculate the amount to reserve and update totals
+    uint32 amountToReserve = uint32(Math.min(mintCount, amountAvailable));
+    claim.total += amountToReserve;
+    _mintDetailsPerWallet[creatorContractAddress][instanceId][msg.sender].reservedCount += amountToReserve;
+    if (amountToReserve > 0 && claim.cost > 0) {
+      _sendFunds(claim.paymentReceiver, claim.cost * amountToReserve);
+    }
+    // Refund any overpayment
+    if (amountToReserve != mintCount) {
+      uint256 refundAmount = msg.value - (claim.cost + MINT_FEE) * amountToReserve;
+      _sendFunds(payable(msg.sender), refundAmount);
+    }
+    emit GachaClaimMintReserved(creatorContractAddress, instanceId, msg.sender, amountToReserve);
+  }
+
+  /**
+   * See {IGachaLazyClaim-deliverMints}.
+   */
+  function deliverMints(IGachaLazyClaim.Mint[] calldata mints) external override {
+    _validateSigner();
+    for (uint256 i; i < mints.length; ) {
+      Mint calldata mintData = mints[i];
+      Claim memory claim = _getClaim(mintData.creatorContractAddress, mintData.instanceId);
+      if (mintData.variationIndex > claim.tokenVariations || mintData.variationIndex < 1)
+        revert IGachaLazyClaim.InvalidVariationIndex();
+
+      // mint parameters
+      uint256[] memory tokenIds = new uint256[](1);
+      tokenIds[0] = claim.startingTokenId + mintData.variationIndex - 1;
+      address[] memory receivers = new address[](mintData.recipients.length);
+      uint256[] memory amounts = new uint256[](mintData.recipients.length);
+      for (uint256 j; j < mintData.recipients.length; ) {
+        address receiver = mintData.recipients[j].receiver;
+        uint256 mintCount = mintData.recipients[j].mintCount;
+        UserMint storage userDetails = _mintDetailsPerWallet[mintData.creatorContractAddress][mintData.instanceId][receiver];
+
+        if (userDetails.deliveredCount + mintCount > userDetails.reservedCount)
+          revert IGachaLazyClaim.CannotMintMoreThanReserved();
+        receivers[j] = receiver;
+        amounts[j] = mintCount;
+        _mintDetailsPerWallet[mintData.creatorContractAddress][mintData.instanceId][receiver].deliveredCount+= uint32(mintCount);
```

Comment:
change this to read:
userDetails.deliveredCount += mintCount

Can reference the storage variable to save gas.  Should not need to cast mintCount since it's already a uint32.  (if it's not change it so it is)